back/spec/uploaders/base_image_uploader_spec.rb

```diff
@@ -28,7 +29,83 @@
       uploader.store!(file)
 
       image = MiniMagick::Image.open(uploader.file.path)
-      expect(image.exif).to be_empty
+
+      allowed_exif_keys = %w[
+        Orientation
+        ResolutionUnit
+        XResolution
+        YCbCrPositioning
+        YResolution
+      ]
+      expect(image.exif.keys - allowed_exif_keys).to be_empty
+
+      exiftool_output = `exiftool #{Shellwords.escape(uploader.file.path)}`
+      # Check the command executed successfully
+      expect($CHILD_STATUS.success?).to be true
+
+      # Tags that are  technical/structural image metadata,
+      # not user-supplied or privacy-sensitive metadata.
+      # The command exiftool -all=, used to strip metadata in BaseImageUploader,
+      # is designed to remove most metadata tags. However, it cannot remove
+      # fundamental file system information or core image characteristics that
+      # are inherent to the file's existence and structure.
+      technical_tags = [
+        'Bits Per Sample',
+        'Color Components',
+        'Directory',
+        'Encoding Process',
+        'Exif Byte Order',
+        'ExifTool Version Number',
+        'File Access Date/Time',
+        'File Inode Change Date/Time',
+        'File Modification Date/Time',
+        'File Name',
+        'File Permissions',
+        'File Size',
+        'File Type',
+        'File Type Extension',
+        'Image Height',
+        'Image Size',
+        'Image Width',
+        'Media White Point',
+        'Megapixels',
+        'MIME Type',
+        'Primary Platform',
+        'Resolution Unit',
+        'X Resolution',
+        'XResolution',
+        'Y Cb Cr Sub Sampling',
+        'Y Cb Cr Positioning',
+        'Y Resolution',
+        'YCbCrPositioning',
+        'YResolution'
+      ]
+
+      present_tags = exiftool_output.lines.map do |l|
+        l.split(':').first.strip
+      end.uniq
+
+      present_tags.reject! do |tag|
+        tag.empty? || tag == uploader.file.path.split('/').last
+      end
+
+      metadata_tags = present_tags - technical_tags
+
+      expect(
+        metadata_tags.reject do |tag|
+          tag.include?('Orientation') ||
+          tag.include?('ICC') ||
+          tag.include?('Profile') ||
+          tag.include?('Curve') ||
+          tag.include?('Matrix') ||
+          tag.include?('Color Space') ||
+          tag.include?('Illuminant') ||
+          tag.include?('CMM') ||
+          tag.include?('Device') ||
+          tag.include?('Rendering') ||
+          tag.include?('Adaptation')
+        end
+      ).to be_empty
```

Comment:
I don't quite understand what we're testing here. Isn't it sufficient to test `image.exif.keys`, or does that contain less information? Testing against the tool's output feels a bit brittle.

If we really need this test, I think it'd be helpful to include an example of the output.